streams/quickstart/java/src/main/resources/archetype-resources/src/main/java/WordCount.java

```diff
@@ -59,17 +63,19 @@ public String apply(String key, String value) {
                        return value;
                    }
                 })
-               .count("Counts")
-               .to(Serdes.String(), Serdes.Long(), "streams-wordcount-output");
+               .count(Materialized.<String, Long, KeyValueStore<Bytes, byte[]>>as("Counts"))
+               .toStream()
+               .to("streams-wordcount-output", Produced.with(Serdes.String(), Serdes.Long()));
 
 
         /* ------- use the code below for Java 8 and uncomment the above ----
 
         builder.stream("streams-plaintext-input")
```

Comment:
The Java 7 code reads:
```java
builder.<String, String>stream("streams-plaintext-input")
```
Why don't we specify types in the Java 8 example? (Or: Do we need the types in the Java 7 example in the first place?)

Reply:
it would be needed in both cases unless we save the stream as a variable of type `KStream<String, String>`